source/server/active_stream_listener_base.h

```diff
@@ -179,5 +180,30 @@ struct ActiveTcpConnection : LinkedObject<ActiveTcpConnection>,
 using ActiveConnectionPtr = std::unique_ptr<ActiveTcpConnection>;
 using ActiveConnectionCollectionPtr = std::unique_ptr<ActiveConnections>;
 
+// The mixin that handles the composition type ActiveConnectionCollection. This mixin
+// provides the connection removal helper and the filter chain removal helper.
+class OwnedActiveStreamListenerBase : public ActiveStreamListenerBase {
```

Comment:
Where is `ActiveStreamListenerBase` used at this point? Why do we need this independent class?

/wait-any

Reply:
Right now there is no usage. It will simplify the listener filter test as a follow up.
I see a quite a few tests have to go though the entire the filter chain manager selection to verify that the listener filter chain allows a socket or reject a socket.

Reply:
OK can you update the comments to indicate this splitting is for tests?

/wait

Reply:
added the comments